### PR TITLE
Accept Any Single Argument

### PR DESCRIPTION
**Pull request by ThibaultLemaire**
Raising an exception turns into an internal error for mypy which is not very user-friendly and is not visible in my IDE.
Also and more importantly an error cannot be ignored via `# type: ignore` which is sure to break any CI pipeline.
By sending a note instead, the user is warned and is free to ignore the error for now and maybe contribute a fix later ;)

### CODE REVIEW

**Comment by Delgan** [comment]
Hi. This makes me wondering: does the user needs to be notified at all? I mean, the `logger` accepts actually objects of any type and will call `str(message)` before logging the message. The only problem is if `args` or `kwargs` are not empty. In this case, Loguru will call `message.format(args, kwargs)` so `message` needs to be a string-like object.

Could:

```python
elif isinstance(log_msg_expr, (IntExpr, FloatExpr)):
    return ctx.default_return_type
```

be generalized to something like the following?

```python
elif not ctx.args[1] and not ctx.args[2]:
    return ctx.default_return_type
```
In the remaining case, if `args` or `kwargs` is not empty, then `log_msg_expr` needs to be a `str` otherwise the call is probably invalid.

**Comment by kornicameister** [comment]
@Delgan actually notes might be useful because they might let someone know that something is missing instead of silencing it.

But...and that's a big but, I think that your suggestion is actually quite good but there's two drawbacks of it from my POV:

1. 
```
some_variable_but_not_str = ... # some variable coming from somewhere
logger.info(some_variable_but_not_str)
```

as unusual ( or not :P ) as it looks, we will get `NameExpr` here as mentioned in #80 (or at least I thinks so).

2. but if that variable is `NameExpr` resolving to `StrExpr` we should have a linter kicking in because that string can be 

```
'Test { } scenarios'
```

which means a call lacks one `*args`.

**Comment by kornicameister** [comment]
@ThibaultLemaire I can't spot it and I am pretty sure an option like that exists (which makes it a bit frustrating) but would you mind checking if notes can be silenced/ignored via `mypy` CLI or via an editor configuration. 

Another idea might be to introduce config handling for plugin and let users decide if they want to see that particular note.

Anyway, sorry for throwing an exception! It's just me :D not liking to hide errors :D

**Comment by ThibaultLemaire** [comment]
@kornicameister I can't find any configuration option for that in mypy's documentation at that moment (mypy's config file is bit poorly documented imo). If we wan't to be let the user silence unwanted warnings without `# type: ignore`ing everything, we'll have to either piggy-back mypy's config file (if that's even easily possible) or have our own config file.

Regarding your concerns, I think @Delgan's proposition is still valid if we take my proposition for #80, which would look something like:

```python
if isinstance(log_msg_expr, (StrExpr, NameExpr)):
  _check_str_format_call(log_msg_expr, ctx)
elif not ctx.args[1] and not ctx.args[2]:
    return ctx.default_return_type
```

**Comment by kornicameister** [comment]
> Don't be. I prefer that to ignoring potential problems.

Awesome.

Regarding, your code proposal. Looks reasonable to me but I'd swap order of checks. Let's make it clear that if there are neither `*args` nor `**kwargs`, plugin has nothing to do. Also if you don't mind and agree please make it so it lands in its own PR. 

Buf before that happens, let's circle back and assume that code modification, from above, to ignore single argument calls (i.e. `loguru.info(1)`) is already on **master** . In such case, can we stay with an exception to  enforce abnormality of the situation where plugin literally has no idea what went wrong?  Perhaps we could land some tests to show that no matter what we put into logger call, everything will be fine, as long this is a sole argument like @Delgan noticed?

**Comment by ThibaultLemaire** [comment]
> Let's make it clear that if there are neither *args nor **kwargs, plugin has nothing to do

Hm.. But that's not true, you've said so yourself:

``` python
from loguru import logger

logger.info('Test {} scenarios')  # We're missing a format argument there, that's probably a mistake!
```

As for the rest I'm not sure I follow, but I think once we settle on a solution for #80 we'll see clearer here.

Also, I'm sorry if I don't understand sometimes and if I put a lot of code examples it's just that I feel like I understand code better. So please feel free to make code suggestions directly on my code or even to push new commits to show what you mean (we can always squash or revert them before merging anyway). Even if you haven't checked your code, even if it doesn't run, even if it's just pseudo code... Sometimes, it's just easier to show than to explain :)

**Comment by kornicameister** [comment]
Right, I mean that example. You are totally right. It's just all that is really complicated TBH and sometimes I get lost :D

**Comment by ThibaultLemaire** [comment]
Erm... Oh no. Looks like fell down the rabbit hole again :man_shrugging: 

So... Yeah. I tried copy pasting @Delgan's suggestion but it didn't work with single arguments
```python
from loguru import logger

logger.debug(1)
```
so I ended up figuring out what was actually going on and refactoring. A bit.

I didn't find necessary to add much more test cases because they're all treated the same way anyway, and I challenge *anyone* to trigger my little edge case note. If you somehow manage to get it with some alien syntax, well... you know what to do :wink:

File: loguru_mypy/__init__.py [hunk 1]
```diff
@@ -38,6 +36,11 @@
     'Named argument of loguru handler is not valid for given message',
     'loguru',
 )
+NOTE_EDGE_CASE: te.Final[str] = (
```
Comment (kornicameister) [inline loguru_mypy/__init__.py:39]: Wonderful :)

**Review by kornicameister — APPROVED** [review]

**Comment by kornicameister** [comment]
Hey @ThibaultLemaire I am approving that. It looks like legitimate fix and the comment is wonderful :)
I will maybe refactor that code later on, if I manage to find time and that is a luxury these days.

**Comment by ThibaultLemaire** [comment]
Hey @kornicameister, sorry to bother you, but I'd like to use this in my code, would it be possible to have a 0.0.4 release for April ? :)
